digest/reader.go

```diff
@@ -164,3 +165,80 @@ func (r *fdWithDigestContentsReader) ReadDigest() (uint32, error) {
 	}
 	return r.digestBuf.ReadDigest(), nil
 }
+
+// ReaderWithDigest is a reader that that calculates a digest
```

Comment:
Any chance of code sharing with `FdWithDigestReader`? The implementation here looks very similar.

Reply:
+1, could FdWithDigestReader use ReaderWithDigest under the hood?

Reply:
Sounds good, done.